packages/builders/src/components/v2/MediaGallery.ts

```diff
@@ -0,0 +1,118 @@
+
+export interface MediaGalleryBuilderData extends Partial<Omit<APIMediaGalleryComponent, 'items'>> {
+	items: MediaGalleryItemBuilder[];
+}
+
+export class MediaGalleryBuilder extends ComponentBuilder<APIMediaGalleryComponent> {
+	protected readonly data: MediaGalleryBuilderData;
+
+	/**
+	 * Creates a new media gallery from API data.
+	 *
+	 * @param data - The API data to create this container with
+	 * @example
+	 * Creating a media gallery from an API data object:
+	 * ```ts
+	 * const mediaGallery = new MediaGalleryBuilder({
+	 * 	items: [
+	 * 		{
+	 * 			description: "Some text here",
+	 * 			media: {
+	 * 				url: 'https://cdn.discordapp.com/embed/avatars/2.png',
+	 * 			},
+	 * 		},
+	 * 	],
+	 * });
+	 * ```
+	 * @example
+	 * Creating a media gallery using setters and API data:
+	 * ```ts
+	 * const mediaGallery = new MediaGalleryBuilder({
+	 * 	items: [
+	 * 		{
+	 * 			description: "alt text",
+	 * 			media: {
+	 * 				url: 'https://cdn.discordapp.com/embed/avatars/5.png',
+	 * 			},
+	 * 		},
+	 * 	],
+	 * })
+	 * 	.addItems(item2, item3);
+	 * ```
+	 */
+	public constructor(data: Partial<APIMediaGalleryComponent> = {}) {
+		super();
+		this.data = {
+			items: data?.items?.map((item) => new MediaGalleryItemBuilder(item)) ?? [],
+			type: ComponentType.MediaGallery,
+		};
+	}
+
+	/**
+	 * The items in this media gallery.
+	 */
+	public get items(): readonly MediaGalleryItemBuilder[] {
+		return this.data.items;
+	}
+
+	/**
+	 * Adds a media gallery item to this media gallery.
+	 *
+	 * @param input - The items to add
+	 */
+	public addItems(
+		...input: RestOrArray<
+			APIMediaGalleryItem | MediaGalleryItemBuilder | ((builder: MediaGalleryItemBuilder) => MediaGalleryItemBuilder)
+		>
+	): this {
+		const normalized = normalizeArray(input);
+		const resolved = normalized.map((item) => resolveBuilder(item, MediaGalleryItemBuilder));
+
+		this.data.items.push(...resolved);
+		return this;
+	}
+
+	/**
+	 * Removes, replaces, or inserts media gallery items for this media gallery.
+	 *
+	 * @param index - The index to start removing, replacing or inserting items
+	 * @param deleteCount - The amount of items to remove
+	 * @param items - The items to insert
+	 */
+	public spliceItems(
+		index: number,
+		deleteCount: number,
+		...items: RestOrArray<
+			APIMediaGalleryItem | MediaGalleryItemBuilder | ((builder: MediaGalleryItemBuilder) => MediaGalleryItemBuilder)
+		>
+	) {
+		const normalized = normalizeArray(items);
+		const resolved = normalized.map((item) => resolveBuilder(item, MediaGalleryItemBuilder));
+
+		this.data.items.splice(index, deleteCount, ...resolved);
+		return this;
+	}
+
+	/**
+	 * {@inheritDoc ComponentBuilder.toJSON}
+	 */
+	public override toJSON(validationOverride?: boolean): APIMediaGalleryComponent {
+		const { items, ...rest } = this.data;
+
+		const data = {
+			...structuredClone(rest),
+			items: items.map((item) => item.toJSON(validationOverride)),
```

Comment:
```suggestion
			items: items.map((item) => item.toJSON(false)),
```

Whenever sub-items are defined in the root predicate, we pass `false`


https://github.com/discordjs/discord.js/blob/12d2d12262d1195599329b7b1f9e7ceee5696084/packages/builders/src/components/v2/Assertions.ts#L51